frontend/src/views/upload-project/UploadProjectViewMachine.tsx

```diff
@@ -0,0 +1,160 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { assign, Machine } from 'xstate';
+
+export interface UploadProjectViewStateSchema {
+  states: {
+    toast: {
+      states: {
+        hidden: {};
+        visible: {};
+      };
+    };
+    uploadProjectView: {
+      states: {
+        idle: {};
+        unauthorized: {};
+        fileSelected: {};
+        uploading: {};
+        success: {};
+      };
+    };
+  };
+}
+export type SchemaValue = {
+  uploadProjectView: 'idle' | 'unauthorized' | 'fileSelected' | 'success';
+  toast: 'visible' | 'hidden';
+};
+
+export interface UploadProjectViewContext {
+  file: any;
+  message: string;
+  newProjectId: string;
+}
+
+export type ShowToastEvent = { type: 'SHOW_TOAST'; message: string };
+export type HideToastEvent = { type: 'HIDE_TOAST' };
+export type HandleSelectedFileEvent = { type: 'HANDLE_SELECTED_FILE'; file: string };
+export type HandleUploadEvent = { type: 'HANDLE_UPLOAD' };
+export type HandleResponseEvent = { type: 'HANDLE_RESPONSE'; response: any };
+export type UploadProjectEvent =
+  | ShowToastEvent
+  | HideToastEvent
+  | HandleSelectedFileEvent
+  | HandleUploadEvent
+  | HandleResponseEvent;
+
+export const uploadProjectMachine = Machine<UploadProjectViewContext, UploadProjectViewStateSchema, UploadProjectEvent>(
+  {
+    id: 'uploadProjectView',
+    type: 'parallel',
+    context: {
+      file: null,
+      message: null,
+      newProjectId: undefined,
+    },
```

Comment:
Stay consistent between null and undefined, let's go with null since we don't want to tell that the value does not exists but instead that it exists but has no value.